include/clangmetatool/include_graph_dependencies.h

```diff
@@ -45,7 +46,39 @@ struct IncludeGraphDependencies {
    */
   static std::set<clangmetatool::types::FileUID>
   liveDependencies(const clangmetatool::collectors::IncludeGraphData *data,
-                   const clangmetatool::types::FileUID &headerFUID);
+                   const clangmetatool::types::FileUID &fileFUID);
+
+  /*
+   * A data structure for include graph weak dependencies analysis
+   * for a specific source file
+   *
+   * Key: all headers the file directly depends on
+   * Value: a set of direct included headers that could allow the file to
+   * access the key header
+   *
+   * Example:
+   * { "def1.h": {"a.h", "b.h"},
+   *   "def2.h": {"a.h", "c.h"} }
+   *
+   * So that given the example data above it means current analyzing file:
+   * - depends on definitions from "def1.h", "def2.h" directly
+   * - includes "a.h", "b.h", "c.h"
+   * - can access "def1.h" by "a.h" and "b.h"
+   * - can access "def2.h" by "a.h" and "c.h"
+   */
+  typedef std::map<clangmetatool::types::FileUID,
+                   std::set<clangmetatool::types::FileUID>> DirectDependenciesMap;
+
+  /**
+   * Get the 'live' weak dependencies of a header within the given include graph.
+   * This will return data structure(called DirectDependenciesMap) shows weak dependencies of headers
+   *
+   * (See docstring of DirectDependenciesMap typedef above for more details of the data structure)
```

Comment:
```suggestion
   * (See \ref DirectDependenciesMap above for more details of the data structure
```

Reply:
I think this parenthetical sentence can be taken out, isn't it implicit in the prototype? If you want to see the structure of the return type of a function, look at its declaration.